examples/dao/test/dao-app-flow.js

```diff
@@ -0,0 +1,456 @@
+const { getProgram, convert } = require('@algo-builder/algob');
+const {
+  Runtime, AccountStore
+} = require('@algo-builder/runtime');
+const { types, parsing } = require('@algo-builder/web');
+const { assert } = require('chai');
+const { ProposalType, Vote } = require('../scripts/run/common/common');
+const {
+  now, getAddProposalTx, getDepositVoteTokenTx, getWithdrawVoteDepositTx,
+  getClearVoteRecordTx, getClearProposalTx
+} = require('../scripts/run/common/tx-params');
+
+const minBalance = 10e6; // 10 ALGO's
+const initialBalance = 200e6;
+
+/**
+ * Test for scripts flow. Steps:
+ * + create DAO (setup)
+ * + add proposal
+ * + deposit vote tokens
+ * + register vote
+ * + execute proposal
+ * + withdraw vote deposit
+ * + clear vote record
+ * + clear proposal
+ */
+describe('DAO test', function () {
+  const master = new AccountStore(1000e6);
+  let creator = new AccountStore(initialBalance);
+  let proposer = new AccountStore(initialBalance);
+  let voterA = new AccountStore(initialBalance);
+  let voterB = new AccountStore(initialBalance);
+  let depositLsigAcc = new AccountStore(initialBalance); // runtime.account of depositLsig.address()
+  let daoFundLsigAcc = new AccountStore(initialBalance);
+  let proposalLsigAcc = new AccountStore(initialBalance);
+
+  let runtime;
+  let flags;
+  let applicationID;
+  let govTokenID;
+  let depositLsig;
+  let daoFundLsig;
+  let proposalLsig;
+
+  this.beforeAll(async function () {
+    runtime = new Runtime([
+      master, creator, proposer, voterA, voterB,
+      depositLsigAcc, daoFundLsigAcc, proposalLsigAcc
+    ]);
+
+    flags = {
+      sender: creator.account,
+      localInts: 9,
+      localBytes: 7,
+      globalInts: 4,
+      globalBytes: 2
+    };
+  });
+
+  const getGlobal = (key) => runtime.getGlobalState(applicationID, key);
+
+  // fetch latest account state
+  function syncAccounts () {
+    // [
+    //   creator, proposer, voterA, voterB,
+    //   depositLsigAcc, daoFundLsigAcc, proposalLsigAcc
+    // ].map((acc) => runtime.getAccount(acc.address));
```

Comment:
can we use this map ?

Reply:
wanted to, instead of repeating code. But this doesn't update the original accounts.

Reply:
could you please check if 
```
mp = []
mp = mp.map((acc......)
```
this works ? If not please remove the comments

Reply:
`map` should not be used. You don't want to create a new list. Please use `for ... of` or `foreach`.

Reply:
Initially i was using `for(..)` loop only, but it wasn't updating the original account.
```js
const accts = [
  creator, proposer, voterA, voterB,
  depositLsigAcc, daoFundLsigAcc, proposalLsigAcc
];
for (let acc of accts) {
  acc = runtime.getAccount(acc.address);
}
```

Reply:
OK, then let's keep what you have.